src/ResourceManager/Sql/Commands.Sql/Auditing/Cmdlet/SetAzureSqlDatabaseAuditingPolicy.cs

```diff
@@ -148,16 +161,16 @@ private void ApplyUserInputToTableAuditingModel(DatabaseAuditingPolicyModel mode
                 ModelAdapter.ClearStorageDetailsCache();
             }
             if (!string.IsNullOrEmpty(StorageKeyType))
```

Comment:
You have this marked as [ValidateNotNullOrEmpty] if you want to check to see if the parameter was populated you should use: this.MyInvocation.BoundParameters.ContainsKey("<param_name>"))